WiserTaskScheduler/WiserTaskScheduler/Modules/GenerateFiles/Services/GenerateFileService.cs

```diff
@@ -141,6 +149,40 @@ private async Task<JObject> GenerateFile(GenerateFileModel generateFile, List<in
                     Html = body
                 };
                 pdfFile = await htmlToPdfConverterService.ConvertHtmlStringToPdfAsync(pdfSettings);
+                
+                // Merge PDF's to generated PDF
+                if (generateFile.Body.MergePdfs.Where(p => !String.IsNullOrEmpty(p.WiserItemId)).ToArray().Length > 0)
+                {
+                    // Make stream of byte array
+                    var stream = new MemoryStream(pdfFile.FileContents);
+                    
+                    //  Create the merge result PDF document
+                    var mergeResultPdfDocument = new Document(stream);
+
+                    // Automatically close the merged documents when the document resulted after merge is closed
+                    mergeResultPdfDocument.AutoCloseAppendedDocs = true;
+
+                    // Set license key received after purchase to use the converter in licensed mode
+                    // Leave it not set to use the converter in demo mode
+                    mergeResultPdfDocument.LicenseKey = GclSettings.Current.EvoPdfLicenseKey;
+
+                    foreach (var pdf in generateFile.Body.MergePdfs)
+                    {
+                        if (String.IsNullOrEmpty(pdf.WiserItemId))
+                        {
+                            continue;
+                        }
+
+                        // Create a scope to get the string wiserItems service. It's needed to get the template
+                        var wiserItemsService = scope.ServiceProvider.GetRequiredService<IWiserItemsService>();
+                        var wiserItemFile = await wiserItemsService.GetItemFileAsync(Convert.ToUInt64(pdf.WiserItemId), "item_id", pdf.PropertyName);
+                        mergeResultPdfDocument.AppendDocument(new Document(new MemoryStream(wiserItemFile.Content)));
+                    }
+
+                    var saveStream = new MemoryStream();
```

Comment:
You should always put a `MemoryStream` in a `using` block (or dispose it in a try/finally), otherwise you might cause memory leaks and crashes.

Reply:
Added 'using' to the declaration, according to https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using